www/src/components/ReactPlayground.js

```diff
@@ -256,11 +267,39 @@ const propTypes = {
 
 function Playground({ codeText, exampleClassName, showCode = true }) {
   // Remove Prettier comments and trailing semicolons in JSX in displayed code.
+  const [copyStatus, setCopy] = useState('Copy to clipboard')
+
   const code = codeText
     .replace(PRETTIER_IGNORE_REGEX, '')
     .trim()
     .replace(/>;$/, '>');
 
+  const [codeToCopy, setcodeToCopy] = useState(code)
+  const handleCodeChange = (e) => { setcodeToCopy(e.target.value) }
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(codeToCopy) //copies code to clipboard
+      .then(setCopy('Copied!'))
+
+  }
+
+
+  const UpdatingPopover = React.forwardRef(
```

Comment:
Components shouldn't be nested within another component.  This should be declared outside.